### PR TITLE
Remove BUMP MPI calls

### PR DESCRIPTION
**Pull request by benjaminmenetrier**
See https://github.com/JCSDA/saber/pull/28#issuecomment-535855089 for details.

I have reproduced what is done for the "eckit::mpi::maxloc" value, but in another class (Info instead of Operation).

### CODE REVIEW

File: src/eckit/mpi/Info.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace eckit {
+namespace mpi {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+struct Info {
+
+    enum Code {
+        INFO_NULL = MPI_INFO_NULL,
+        MAX_INFO_CODE,
```
Comment (mmiesch) [inline src/eckit/mpi/Info.h]: Where is ```MAX_INFO_CODE``` defined?

**Comment by mmiesch** [comment]
I am getting an error on my Mac compiling this with gnu 7.3 + openmpi.  

~~~~~
In file included from /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/eckit/src/eckit/mpi/Info.h:14:0,
                 from /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/eckit/src/eckit/mpi/Comm.h:25,
                 from /Users/miesch/JEDI/code/working_copy/src/ufo-bundle/eckit/src/eckit/mpi/Comm.cc:11:
/opt/modules/gnu-7.3.0/openmpi/3.1.2/include/mpi.h:311:74: error: enumerator value for 'INFO_NULL' must have integral or unscoped enumeration type
 #define OMPI_PREDEFINED_GLOBAL(type, global) ((type) ((void *) &(global)))
                                                                          ^
/opt/modules/gnu-7.3.0/openmpi/3.1.2/include/mpi.h:747:23: note: in expansion of macro 'OMPI_PREDEFINED_GLOBAL'
 #define MPI_INFO_NULL OMPI_PREDEFINED_GLOBAL(MPI_Info, ompi_mpi_info_null)
                       ^~~~~~~~~~~~~~~~~~~~~~
/Users/miesch/JEDI/code/working_copy/src/ufo-bundle/eckit/src/eckit/mpi/Info.h:24:21: note: in expansion of macro 'MPI_INFO_NULL'
         INFO_NULL = MPI_INFO_NULL,
                     ^~~~~~~~~~~~~
~~~~~

**Comment by benjaminmenetrier** [comment]
I have simplified the code to get access to MPI_INFO_NULL, removing enum and MAX_INFO_CODE. The error you get should be gone. Let me know.

**Comment by mmiesch** [comment]
It looks like the reason it's failing the Travis test is because of our old friend ```mpi.h```.  Apparently ecbuild is not finding mpi.

File: src/eckit/mpi/Info.cc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
```
Comment (ytremolet) [inline src/eckit/mpi/Info.cc]: Please change the copyright.

File: src/eckit/mpi/Info.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
```
Comment (ytremolet) [inline src/eckit/mpi/Info.h]: Same here.

**Comment by benjaminmenetrier** [comment]
> It looks like the reason it's failing the Travis test is because of our old friend `mpi.h`. Apparently ecbuild is not finding mpi.

Thanks @mmiesch for spotting this issue. In eckit, `mpi.h` is loaded in `Parallel.h` only (not in `Serial.h` of course), which means that:
 - the call to MPI functions has to be done in `Parallel.h`.
 - returning `MPI_INFO_NULL` has to be a method of the communicator.

As a consequence:
 - `Info.h` and `Info.cc` have been removed.
 - `Comm.h`, `Parallel.h/cc` and `Serial.h/cc` have a new method `infoNull()` that returns MPI_INFO_NULL (for the parallel case) or the largest possible integer (for Serial), inspired by what is done for the method `anyTag()`.

The Travis tests are now passing.

**Comment by mmiesch** [comment]
Great - thanks @benjaminmenetrier for fixing that - I guess the reason it worked in my tests is that I generally set the ```CC``` and ```CXX``` environment variables equal to their mpi wrappers, e.g. ```mpicc``` and ```mpicxx``` for OpenMPI.

**Pull request merged**